test/FullRange.t.sol

```diff
@@ -0,0 +1,453 @@
+        token2.approve(address(modifyPositionRouter), type(uint256).max);
+        token0.approve(address(swapRouter), type(uint256).max);
+        token1.approve(address(swapRouter), type(uint256).max);
+        token2.approve(address(swapRouter), type(uint256).max);
+        token0.approve(address(manager), type(uint256).max);
+        token1.approve(address(manager), type(uint256).max);
+        token2.approve(address(manager), type(uint256).max);
+    }
+
+    function testBeforeInitializeAllowsPoolCreation() public {
+        vm.expectEmit(true, true, true, true);
+        emit Initialize(id, key.currency0, key.currency1, key.fee, key.tickSpacing, key.hooks);
+        snapStart("initialize");
+        manager.initialize(key, SQRT_RATIO_1_1);
+        snapEnd();
+
+        (, address liquidityToken) = fullRange.poolInfo(id);
+
+        assertFalse(liquidityToken == address(0));
+    }
+
+    function testBeforeInitializeRevertsIfWrongSpacing() public {
+        PoolKey memory wrongKey =
+            PoolKey(Currency.wrap(address(token0)), Currency.wrap(address(token1)), 0, TICK_SPACING + 1, fullRange);
+
+        vm.expectRevert(FullRange.TickSpacingNotDefault.selector);
+        manager.initialize(wrongKey, SQRT_RATIO_1_1);
+    }
+
+    function testInitialAddLiquiditySucceeds() public {
+        manager.initialize(key, SQRT_RATIO_1_1);
+
+        uint256 prevBalance0 = MockERC20(token0).balanceOf(address(this));
+        uint256 prevBalance1 = MockERC20(token1).balanceOf(address(this));
+
+        address token0Addr = address(token0);
+        address token1Addr = address(token1);
+
+        snapStart("add liquidity");
+        fullRange.addLiquidity(token0Addr, token1Addr, 3000, 10 ether, 10 ether, address(this), MAX_DEADLINE);
+        snapEnd();
+
+        (bool owed, address liquidityToken) = fullRange.poolInfo(id);
+
+        assertEq(MockERC20(token0).balanceOf(address(this)), prevBalance0 - 10 ether);
+        assertEq(MockERC20(token1).balanceOf(address(this)), prevBalance1 - 10 ether);
+
+        assertEq(UniswapV4ERC20(liquidityToken).balanceOf(address(this)), 10 ether - 1000);
+        assertEq(owed, false);
+    }
+
+    function testInitialAddLiquidityFuzz(uint256 amount) public {
+        manager.initialize(key, SQRT_RATIO_1_1);
+
+        if (amount <= 1000) {
+            vm.expectRevert("Input amount does not minimum liquidity");
+            fullRange.addLiquidity(address(token0), address(token1), 3000, amount, amount, address(this), MAX_DEADLINE);
+        } else if (amount >= 10 ** 34) {
+            vm.expectRevert();
+            fullRange.addLiquidity(address(token0), address(token1), 3000, amount, amount, address(this), MAX_DEADLINE);
+        } else {
+            fullRange.addLiquidity(address(token0), address(token1), 3000, amount, amount, address(this), MAX_DEADLINE);
+
+            (bool owed,) = fullRange.poolInfo(id);
+            assertEq(owed, false);
```

Comment:
wouldn't this be false for nil pools? could check pool manager has the pool initialized, otherwise this just checks no revert happened (which I guess is fine)